fraction.rb

```diff
@@ -1,17 +1,64 @@
 class Fraction
   def initialize(numerator, denominator)
-    if !(numerator.is_a?(Integer) && denominator.is_a?(Integer))
-      # TODO: create a proper exception for that issue
-      # TODO: handle case when one or both parameters are not Integer but could be easily transformed into it
-      # by using #to_i
-      raise 'At least some arguments are incorrect: ' + numerator.inspect + ', ' + denominator.inspect
+    if ! numerator.is_a?(Integer) && denominator.is_a?(Integer)
+      # TODO: Proper execption for this issue
+      # TODO: Handle case when one or both parameters are not int but could be easily transformed to int
+      raise 'these are supposed to be numbers: '+numerator.inspect+' '+ denominator.inspect
     end
     @numerator = numerator
     @denominator = denominator
   end
+
+  def numerator
+    @numerator
+  end
+
+  def denominator
+    @denominator
+  end
+
+  def simplify
+
+=begin
+     if @numerator > @denominator then b=@denominator else b=@numerator
+    end
+    end
+
+    while b>1
+     if (@numerator%b == 0 and @denominator%b == 0)
+       @denominator = @denominator /b
+       @numerator = @numerator /b
+     end
+      b=b-1
+=end
+    Fraction.new(@numerator / @numerator.gcd(@denominator), @denominator / @denominator.gcd(@numerator))
+  end
+
+  def view
+      puts "#{@numerator}/#{@denominator}"
```

Comment:
Do not use puts here. The purpose of that method to prepare it for showing. You should not call `puts` right now, only in a place where you will need it. Like:
```ruby
f = Fraction.new(1, 3)
puts f.view
```